azure-pipelines.yml

```diff
@@ -0,0 +1,83 @@
+# File: azure-pipelines.yml
+
+trigger:
+  batch: true
+  branches:
+    include:
+      - main
+      - release/*
+
+resources:
+  repositories:
+    - repository: niveristand-custom-device-build-tools
+      type: github
+      ref: main
+      endpoint: ni (2)
+      name: ni/niveristand-custom-device-build-tools
+# TEMPORARY, include fake custom device to use templates until migrating the templates into build tools makes sense
+    - repository: imitation-cd
+      type: github
+      ref: main
+      endpoint: papowerNI
+      name: papowerNI/imitation-cd
+
+pool:
+  name: custom-device-temp
+  demands:
+    - agent.os -equals Windows_NT
+
+# --------- Build Parameters here ---------
+parameters:
+# Custom Device Name
+  - name: customDevice
+    type: string
+    default: niveristand-slsc-eds-custom-device
+# LabVIEW Versions to Build
+  - name: lvVersions
+    type: object
+    default:
+      - '2020'
+      - '2021'
+      - '2023'
+# Custom Device Build Steps
+  - name: buildSteps
+    type: object
+    default:
+      # (project location, LabVIEW CLI Build operation, Target, Build Specification Name)
+      - ['Source\SLSC EDS.lvproj', 'ExecuteBuildSpec', 'My Computer', 'Configuration Release']
```

Comment:
I'm curious if there is a way we can represent these as named items, rather than a list. I can imagine us wanting to add more over time, or provide default values, and this current pattern could make that challenging.

Reply:
Agreed.  I have been looking at the data structures that can be used in Azure pipelines parameters to see if I can think of another method that ends up cleaner than this.  I think that the "each" method is the only viable way to cycle through each build spec without copying yaml significantly, and "each" iterates through an array like a for loop.  I haven't found another way to pass an array into that iteration yet.

Reply:
Updated to use a map instead of an array, apparently everything is a map in azure pipelines and you can just make your own as you like